mmv1/third_party/terraform/tpgresource/annotations.go

```diff
@@ -52,6 +52,16 @@ func SetMetadataAnnotationsDiff(_ context.Context, d *schema.ResourceDiff, meta
 		return nil
 	}
 
+	// Fix the bug that the computed and nested "annotations" field disappears from the terraform plan.
+	// https://github.com/hashicorp/terraform-provider-google/issues/17756
+	// The bug is introduced by SetNew on "metadata" field with the object including "effective_annotations".
+	// "effective_annotations" cannot be set directly due to a bug that SetNew doesn't work on nested fields
+	// in terraform sdk.
+	// https://github.com/hashicorp/terraform-plugin-sdk/issues/459
+	if !d.GetRawPlan().GetAttr("metadata").AsValueSlice()[0].GetAttr("annotations").IsWhollyKnown() {
+		return nil
```

Comment:
At creation time this works because we'll keep the `effective_annotations` value unknown (/(known after apply) as it appears to end users) as a result of this, right?

Reply:
At creation time this works because we'll keep the `annotations` value unknown (/(known after apply) as it appears to end users) as a result of this. Otherwise, `annotations` doesn't appear in the terraform plan.

I tested with the `labels` field
Before the fix, in terraform plan
```
  + resource "google_cloud_run_service" "default" {
      + metadata {
          + effective_annotations = (known after apply)
          + effective_labels      = (known after apply)
          + generation            = 0
          + namespace             = "terraform-dev-zhenhuali"
          + resource_version      = (known after apply)
          + self_link             = (known after apply)
          + terraform_labels      = (known after apply)
          + uid                   = (known after apply)
        }

```

With the fix, in terraform plan
```
  + resource "google_cloud_run_service" "default" {
      + metadata {
          + effective_annotations = (known after apply)
          + effective_labels      = (known after apply)
          + generation            = (known after apply)
          + labels                = (known after apply)
          + namespace             = "terraform-dev-zhenhuali"
          + resource_version      = (known after apply)
          + self_link             = (known after apply)
          + terraform_labels      = (known after apply)
          + uid                   = (known after apply)
        }
```

Reply:
Oh, the failure was on labels itself. Got it, thanks!